sale_order_global_stock_route/models/sale_order.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    route_id = fields.Many2one(
+        comodel_name="stock.location.route",
+        string="Route",
+        domain=[("sale_selectable", "=", True)],
+        help="When you change this field all the lines will be changed."
+        " After use it you will be able to change each line.",
+    )
+
+    @api.onchange("route_id")
+    def _onchange_route_id(self):
+        """We could do sale order line route_id field compute store writable.
+        But this field is created by Odoo so I prefer not modify it.
+        """
+        for line in self.order_line:
```

Comment:
As onchange is always applied on a single record (see https://github.com/odoo/odoo/blob/15.0/odoo/models.py#L6489)

, I would do:

```suggestion
        self.order_line.update({'route_id': self.order_id.route_id})
```

Reply:
I didn't know it worked like this, thanks for the lesson learned.

Reply:
@stefan-tecnativa No problem. The most performances gain can be earned when you have several attributes to modify as you can do `recordset.update({<attribute 1>: value, <attribute 2>: value, ...})`

Reply:
For the present case, I'd go for a direct attribution. It makes no difference at ORM level and the code is much more expresive:
```python
        self.order_line.route_id = self.route_id
```

Reply:
I am agree with you.